feat(docs): combine using libraries docs (#18973)
expo/expo: docs/pages/workflow/using-libraries.md

title: Using libraries
---

import { ConfigReactNative } from '~/components/plugins/ConfigSection';
import PlatformsSection from '~/components/plugins/PlatformsSection';
import InstallSection from '~/components/plugins/InstallSection';
import { Terminal } from '~/ui/components/Snippet';
import SnackInline from '~/components/plugins/SnackInline';

> Every app is made up of some combination of React Native libraries. It's important to understand how to determine whether a third-party library will work in your project or not.

Reviewer: ```suggestion
Every app is inevitably going to use a third-party library, so it's important to understand how to determine whether a library is compatible with your project.
```

}
```

</SnackInline>

## Using Expo SDK Libraries

The Expo SDK picks up where the React Native core libraries end - it provides access to a lot of useful device and system functionality like audio, barcode scanning, camera, calendar, contacts, video, and so on. It also adds other powerful libraries like updates, maps, OAuth authentication tools, and more. For details on how we decide what goes into the Expo SDK, [read here](https://expo.fyi/whats-in-the-sdk).

To use a library from the Expo SDK, find the one you are looking for in the [API Reference](/versions/latest/) or through the documentation Search bar.

<ConfigReactNative>

> If you initialized your app using `npx react-native init` and you don't have the `expo` package installed in it yet, please refer to the guide for [installing Expo modules guide](/bare/installing-expo-modules). It should take about five minutes to configure in an existing app.

Reviewer: ```suggestion
If you initialized your app using `npx react-native init` and you don't have the `expo` package installed in it yet, refer to the [installing Expo modules guide](/bare/installing-expo-modules).
```

### Installing a Third-Party Library

> We recommend always using `npx expo install` instead of `npm install` or `yarn add` directly because it allows [Expo CLI][expo-cli] to pick a compatible version of a library when possible and also warn you about known incompatibilities.

Once you have determined if the library is compatible with React Native, use [Expo CLI][expo-cli] to install the package:

<Terminal cmd={['$ npx expo install @react-navigation/native']} />

Be sure to follow the project website or README for any additional configuration and usage instructions. You can get to the README quickly using this command:

<Terminal cmd={['$ npx npm-home @react-navigation/native']} />

If the module needs additional native configuration, you can do so using [config plugins](/guides/config-plugins). Some packages require a config plugin but they don't have one yet, you can refer to the list of [out-of-tree config plugins](https://github.com/expo/config-plugins/).

<ConfigReactNative>

> If your project does not support [Expo Prebuild](/workflow/prebuild) then you cannot use [config plugins](/guides/config-plugins). You can either [adopt Expo Prebuild](/guides/adopting-prebuild) or setup and configure each library manually by following any additional setup guides in the respective module's npm page.

Reviewer: ```suggestion
If your project does not support [Expo Prebuild](/workflow/prebuild) then you won't be able to use [config plugins](/guides/config-plugins). You can either [adopt Expo Prebuild](/guides/adopting-prebuild) or setup and configure each library manually by following any additional setup guides from the respective module's website or README.
```

i'm not crazy about this wording - "does not support Expo Prebuild" -- but let's try it, given that it's already scoped within a ConfigReactNative block
